test/jdk/java/lang/StrictMath/FdlibmTranslit.java

```diff
@@ -808,4 +821,188 @@ else if(hx < 0x3c900000) {      /* when |x|<2**-54, return x */
+            }
+
+            /* |x| in [22, log(maxdouble)] return 0.5*exp(|x|) */
+            if (ix < 0x40862E42)  return h*StrictMath.exp(Math.abs(x)); // TODO switch to translit
+
+            /* |x| in [log(maxdouble), overflowthresold] */
+            // lx = *( (((*(unsigned*)&one)>>29)) + (unsigned*)&x);
+            // lx =  (((*(unsigned*)&one)>>29)) + (unsigned*)&x ;
+            lx = __LO(x);
+            if (ix<0x408633CE || ((ix==0x408633ce)&&(Long.compareUnsigned(lx, 0x8fb9f87d) <= 0 ))) {
+                w = StrictMath.exp(0.5*Math.abs(x)); // TODO switch to translit
+                t = h*w;
+                return t*w;
+            }
+
+            /* |x| > overflowthresold, sinh(x) overflow */
+            return x*shuge;
+        }
+    }
+
+    /**
+     * Method :
+     * mathematically cosh(x) if defined to be (exp(x)+exp(-x))/2
+     *      1. Replace x by |x| (cosh(x) = cosh(-x)).
+     *      2.
+     *                                                      [ exp(x) - 1 ]^2
+     *          0        <= x <= ln2/2  :  cosh(x) := 1 + -------------------
+     *                                                         2*exp(x)
+     *
+     *                                                exp(x) +  1/exp(x)
+     *          ln2/2    <= x <= 22     :  cosh(x) := -------------------
+     *                                                        2
+     *          22       <= x <= lnovft :  cosh(x) := exp(x)/2
+     *          lnovft   <= x <= ln2ovft:  cosh(x) := exp(x/2)/2 * exp(x/2)
+     *          ln2ovft  <  x           :  cosh(x) := huge*huge (overflow)
+     *
+     * Special cases:
+     *      cosh(x) is |x| if x is +INF, -INF, or NaN.
+     *      only cosh(0)=1 is exact for finite x.
+     */
+    static class Cosh {
+        private static final double one = 1.0, half=0.5, huge = 1.0e300;
+        private static double compute(double x) {
+            double t,w;
+            int ix;
+            /*unsigned*/ int lx;
+
+            /* High word of |x|. */
+            ix = __HI(x);
+            ix &= 0x7fffffff;
+
+            /* x is INF or NaN */
+            if(ix>=0x7ff00000) return x*x;
+
+            /* |x| in [0,0.5*ln2], return 1+expm1(|x|)^2/(2*exp(|x|)) */
+            if(ix<0x3fd62e43) {
+                t = expm1(Math.abs(x));
+                w = one+t;
+                if (ix<0x3c800000) return w;        /* cosh(tiny) = 1 */
+                return one+(t*t)/(w+w);
+            }
+
+            /* |x| in [0.5*ln2,22], return (exp(|x|)+1/exp(|x|)/2; */
+            if (ix < 0x40360000) {
+                t = StrictMath.exp(Math.abs(x)); // TODO switch to translit
+                return half*t+half/t;
+            }
+
+            /* |x| in [22, log(maxdouble)] return half*exp(|x|) */
+            if (ix < 0x40862E42)  return half*StrictMath.exp(Math.abs(x)); // TODO switch to translit
+
+            /* |x| in [log(maxdouble), overflowthresold] */
+            lx = __LO(x);
```

Comment:
Same concern as above about the transliteration to `__LO(x)`

Reply:
Right; sinh and cosh overflow at the same input.